pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -151,10 +153,11 @@ func (r *ReconcileMasterUserRecord) ensureUserAccount(log logr.Logger, recAccoun
 		recordSpecUserAcc: recAccount,
 	}
 	if err := sync.synchronizeSpec(); err != nil {
-		return err
+		return r.wrapErrorWithStatusUpdate(log, record, r.setStatusFailed(unableToSynchronizeUserAccountSpecReason), err,
+			"update of the UserAccount.spec in the cluster '%s' failed", recAccount.TargetCluster)
 	}
 	if err := sync.synchronizeStatus(); err != nil {
-		return err
+		return errs.Wrapf(err, "update of the UserAccount.status in the cluster '%s' failed", recAccount.TargetCluster)
```

Comment:
As for the error msg. Yes, you are right - I used copy-paste driven development and only changed spec to status, but didn't realize that the error msg should be different - fixed: 4db5f75

The reason why I don't update the status is that synchronization of the statuses shouldn't have any impact on the overall functionality. In addition, if the status update of the MasterUserRecord fails while synchronizing with UserAccount, there is a very low probability that it would pass in this case. This is the reason why I don't see updating the status to "not ready" not useful.

Reply:
But in this case we could end up with overall status not ready (it was not
ready, status sync failed so we didn’t change it ready) but we won’t see
the reason in the status. I didn’t read the code very closely though. It’s
just my guess.

Reply:
Yes, you are right with this - for now, it would be still `updating` or `provisioning` and the overall status would be `not-ready` because the attempt to set it to `ready` failed - I didn't realize it. 

But I'm still reluctant with setting a new status there because I think that the status should reflect the result of meeting the desired spec, not result of synchronizing the status. But what I would do is instead of setting a new reason, I would set only a message 
`status synchronization with UserAccount for targetCluster 'member-cluster' failed)`
and keep the reason as well as the overall status same. Does it make sense to you.

and just BTW - as for your concerns:
> The status sync can fail because of different reasons. For example because we can't get user accounts (member cluster is not available or something else).

The status sync can fail only because of one reason - status update of the MasterUserRecord. The UserAccounts are retrieved prior to doing synchronization: https://github.com/codeready-toolchain/host-operator/pull/23/files#diff-a7167111790d962cde50e8d4ded9aba5R130 in that case the corresponding condition is set

Reply:
Yeah. Sounds good to me.

Reply:
done here 42c7951
it got a bit tricky because:
* I had to tweak the mock in the test to fail only once
* but then when the update succeeded (while setting the error message to the condition) it also updated the embedded UserAccount status as it was still stored in the object (it's passed as a pointer). To solve this, I had to remove the newly added UserAcc.status or bring back original the original values when there is a sync status failure.